### ISSUE TITLE
1.2.x version regression: when `install` called multiple times with different "extras", then "extras" from previous step are removed

### ISSUE THREAD

**Issue by AIGeneratedUsername** [python-poetry/poetry#5895]
# Steps to reproduce (**1.2.x** version) 
1. `poetry install -E foobar`
2. `poetry install -E spameggs`

After calling `-E spameggs`, all extras from "foobar" will be removed.

# Old behavior (**1.1.x** version)

1. `poetry install -E foobar`
2. `poetry install -E spameggs`

After calling `-E spameggs`, all extras from "foobar" are kept. In result, I have all "foobar" and "spameggs" dependencies.

# Problems

- This change is not mentioned in the changelog https://python-poetry.org/history/ 
- This change is Poetry regression / degradation.

# Rationale

In the past, I was able to use Docker multi-stage builds:
```dockerfile
FROM python:3.10-slim-bullseye as base
# Install only main dependencies
RUN poetry install --no-interaction --no-ansi --no-root --no-dev


FROM base as testing
# We have runtime dependencies, now add only "dev" dependencies
RUN poetry install --no-interaction --no-ansi --no-root


FROM base as deploy
# We have runtime dependencies, now add only cloud dependencies
RUN poetry install --no-interaction --no-ansi --no-root --no-dev -E cloud_deploy
```
In other words, I was able to add more and more "extras" on every CI/CD step. Since **1.2.x** version such behavior is broken and I see no workarounds.

**Comment by dimbleby** [comment]
The 1.2 behaviour seems like an improvement to me.  If you want to install more than one set of extras then, per the help

```
  -E (--extras)          Extra sets of dependencies to install. (multiple values allowed)
```
multiple values are allowed

**Comment by AIGeneratedUsername** [comment]
> The 1.2 behaviour seems like an improvement to me. If you want to install more than one set of extras then, per the help
> 
> ```
>   -E (--extras)          Extra sets of dependencies to install. (multiple values allowed)
> ```
> 
> multiple values are allowed

Multiple values were allowed in **1.1.x**. You are talking about a different thing. 

I need to install more and more extras during multiple CI/CD stages. Can you see the difference between installing multiple extras at once and installing multiple extras by calling `poetry install` multiple times? The second option is what I need:
```
CI stage 1:
    poetry install -E base

CI stage 2:
    poetry install -E foobar


CI stage 3:
    poetry install -E cloud_deploy
```

Of course I can always install `-E "foo bar spam eggs"` at the start, but this is not what I need. E.g. I do not need cloud dependencies for my unit tests. I need add cloud deps as extra later, but I need to keep all other extras.

**Comment by dimbleby** [comment]
stage 1 install foo, stage 2 install foo + bar, stage 3 install foo + bar + baz?

**Comment by AIGeneratedUsername** [comment]
> stage 1 install foo, stage 2 install foo + bar, stage 3 install foo + bar + baz?

Yes, exactly. 

I can do as you wrote, but this will result in more boilerplate code inside Dockerfile multi-stages. I will need to copy/paste specific extras again and again.

---

I am fine with the current (new) Poetry behavior, but it must be documented. Of course, if you say that this is a feature and not a bug.

**Comment by neersighted** [comment]
Not a bug / docs are updated by #6432.
